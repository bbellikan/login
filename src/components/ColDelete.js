import React, { Component } from 'react';

class ColDelete extends Component {
    constructor(props) {
        super(props);
        this.deleteRow = this.deleteRow.bind(this);
    }

    deleteRow(rowNumber){
        this.props.deleteRow(rowNumber);
    }

    render() {
        return (
            <div className="grid-col"
                 style={{order: 999, right: 0, position: 'sticky', backgroundColor: 'white'}}
            >
                <div className="cell cell-header"/>
                {this.props.gridData.map((item, index) => {
                    return (
                        <div key={index}
                             onClick = {()=>this.deleteRow(index)}>
                                <i className = "grid-coledit fa fa-trash"/>
                        </div>
                    )
                })}
            </div>
        );
    }
}

export default ColDelete;
