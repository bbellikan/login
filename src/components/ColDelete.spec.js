import React from 'react';
import { shallow } from 'enzyme';
import ColDelete from './ColDelete';

const setup = (setupProps = {}) => {
    const defaultProps = {
        gridData : [{
            type: 'POS',
            extdesc: 'ABC VONS Store 1797 SAN DIEGO CAUS',
            description: 'POS Transaction VONS Store 1797 SAN DIEGO CAUS',
            amount: '-4.43',
            differed: false,
            date: '2016-02-23'
        }],
        deleteRow:jest.fn(),
    };


    const props = {...defaultProps, ...setupProps};
    const wrapper = shallow(
        <ColDelete {...props}/>
    );

    return {
        props,
        wrapper
    };
};

describe('ColDelete', () => {
    it('renders without crashing', () => {
        const {wrapper} = setup();
        expect(wrapper).toMatchSnapshot();
    });

});
