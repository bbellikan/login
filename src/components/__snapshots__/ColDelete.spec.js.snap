// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColDelete renders without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ColDelete
    deleteRow={[Function]}
    gridData={
        Array [
            Object {
              "amount": "-4.43",
              "date": "2016-02-23",
              "description": "POS Transaction VONS Store 1797 SAN DIEGO CAUS",
              "differed": false,
              "extdesc": "ABC VONS Store 1797 SAN DIEGO CAUS",
              "type": "POS",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="cell cell-header"
/>,
        Array [
          <div
            onClick={[Function]}
>
            <i
                        className="grid-coledit fa fa-trash"
            />
</div>,
        ],
      ],
      "className": "grid-col",
      "style": Object {
        "backgroundColor": "white",
        "order": 999,
        "position": "sticky",
        "right": 0,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "cell cell-header",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="grid-coledit fa fa-trash"
/>,
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "grid-coledit fa fa-trash",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="cell cell-header"
/>,
          Array [
            <div
              onClick={[Function]}
>
              <i
                            className="grid-coledit fa fa-trash"
              />
</div>,
          ],
        ],
        "className": "grid-col",
        "style": Object {
          "backgroundColor": "white",
          "order": 999,
          "position": "sticky",
          "right": 0,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "cell cell-header",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <i
              className="grid-coledit fa fa-trash"
/>,
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "grid-coledit fa fa-trash",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
